packages/api/src/routes/ehr/util.ts

```diff
@@ -0,0 +1,54 @@
+import { Request } from "express";
+import { getPatientMappingOrFail } from "../../command/mapping/patient";
+import { EhrSources } from "../../external/ehr/shared";
+
+export const idRegex = "([a-zA-Z0-9\\_\\-\\.])+";
+
+export type PathDetails = {
+  regex: RegExp;
+  paramRegexIndex?: number;
+  queryParamKey?: string;
+};
+
+export function validatePath(req: Request, paths: PathDetails[]): PathDetails {
+  const validPaths = paths.filter(path => path.regex.test(req.path));
+  if (validPaths.length === 0) throw new Error(`Invalid path ${req.path}`);
```

Comment:
Should we throw 404 on this one? Could log, in addition to throwing, to help w/ debugging. But if someone is calling an endpoint that we didn't approve access to, then 404 seems safer.

Reply:
Done.